rustls/src/client/hs.rs

```diff
@@ -171,17 +171,12 @@ struct ExpectServerHelloOrHelloRetryRequest(ExpectServerHello);
 
 pub fn compatible_suite(
     sess: &ClientSessionImpl,
-    resuming_suite: Option<&suites::SupportedCipherSuite>,
+    resuming_suite: &suites::SupportedCipherSuite,
 ) -> bool {
-    match resuming_suite {
-        Some(resuming_suite) => {
-            if let Some(suite) = sess.common.get_suite() {
-                suite.can_resume_to(&resuming_suite)
-            } else {
-                true
-            }
-        }
-        None => false,
+    if let Some(suite) = sess.common.get_suite() {
+        suite.can_resume_to(&resuming_suite)
+    } else {
+        true
```

Comment:
I generally prefer to use this pattern for things like this. It's a little more compact and I find the different outcomes easier to parse visually. Also works great for early return/continue.

```suggestion
    match sess.common.get_suite() {
        Some(suite) => suite.can_resume_to(&resuming_suite),
        None => true,
    }
```

Reply:
Thanks. I updated the PR with this change.